server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -283,6 +287,10 @@ public void innerOnResponse(Result result) {
                             } finally {
                                 executeNext(pendingExecutions, thread);
                             }
+                            ThreadMXBean threadMXBean = (ThreadMXBean) ManagementFactory.getThreadMXBean();
+                            long bytes = threadMXBean.getThreadAllocatedBytes(Thread.currentThread().getId());
```

Comment:
Also, it is worth checking if thread allocation tracking is enabled & supported, for not doing unnecessary work: `ThreadMXBean::isThreadAllocatedMemorySupported()` and `ThreadMXBean::isThreadAllocatedMemoryEnabled()`